platform/darwin/src/MGLNetworkConfiguration_Private.h

```diff
@@ -0,0 +1,25 @@
+#import "MGLNetworkConfiguration.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+extern NSString * const kMGLDownloadPerformanceEvent;
+
+@protocol MGLNetworkEventDelegate <NSObject>
+
+@optional
+- (void)networkConfiguration:(MGLNetworkConfiguration *)networkConfiguration didReceiveNetworkEvent:(NSDictionary *)event;
```

Comment:
Is there a reason why we shouldn't add the event "name" to this delegate method? e.g.

`- (void)networkConfiguration:(MGLNetworkConfiguration *)networkConfiguration didReceiveNetworkEvent:(NSString *)event attributes:(NSDictionary *)eventAttributes`?

@rclee we should consider making events their own type rather than `NSString`